src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -79,6 +81,16 @@ public function isFollowingRedirects()
     }
 
     /**
+     * Sets whether to emulate back/forward browser navigation (skip redirects).
+     *
+     * @param bool $browserNavigation Whether to emulate browser navigation
+     */
+    public function browserNavigation($browserNavigation = true)
```

Comment:
I would not make it configurable as browsers don't make this configurable either. I understand that you probably want to be able to keep BC here. But I don't think that this is a good idea. The current behavior is wrong and we should fix it. What needs to be done though is to add a note in the CHANGELOG/UPGRADE files to document this change.